src/log-analytics-solution/azext_log_analytics_solution/_params.py

```diff
@@ -0,0 +1,49 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from ._validators import validate_workspace_resource_id
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('monitor log-analytics solution create') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help='Name of the log-analytics solution.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type)
+        c.argument('plan_name', help='Name of the plan for solution. For Microsoft published solution it should be in the format of solutionType(workspaceName). '
+                                     'SolutionType part is case sensitive. For third party solution, it can be anything.')
+        c.argument('plan_publisher', help='Publisher name of the plan for solution. For gallery solution, it is Microsoft.')
+        c.argument('plan_product', help='Product name of the plan for solution. '
+                                        'For Microsoft published gallery solution it should be in the format of OMSGallery/<solutionType>. This is case sensitive.')
+        c.argument('workspace_resource_id', options_list=['--workspace-resource-id', '-w'],
```

Comment:
please support workspace name as well

Reply:
I mean use one argument "workspace" that supports both name and resource id as `storage_account` does.

---
In reply to: [415257345](https://github.com/Azure/azure-cli-extensions/pull/1545#discussion_r415257345) [](ancestors = 415257345)